### PR TITLE
Fix bug in reduce_op caused by PR #13534

### PR DESCRIPTION
**Pull request by qingqing01**
Fix https://github.com/PaddlePaddle/Paddle/issues/13741
Fix https://github.com/PaddlePaddle/models/issues/1338

### CODE REVIEW

File: paddle/fluid/operators/cub_reduce.h [hunk 1]
```diff
@@ -293,7 +294,10 @@ void TensorReduce(const framework::Tensor& x, framework::Tensor* y,
   }
   auto x_data = x.data<Tx>();
   auto y_data = y->mutable_data<Ty>(x.place());
-  if (reduce_num == 1) return;
+  if (reduce_num == 1) {
+    framework::TensorCopy(x, y->place(), y);
```
Comment (wanghaoshuang) [inline paddle/fluid/operators/cub_reduce.h:299]: `TensorCopy` will resize y to x.shape. So we should restore the shape of y after `TensorCopy`.

File: python/paddle/fluid/tests/unittests/test_reduce_op.py [hunk 2]
```diff
@@ -243,5 +243,73 @@ def test_check_grad(self):
...
+        }
+
+    def test_check_output(self):
+        self.check_output()
+
+    def test_check_grad(self):
+        self.check_grad(['X'], 'Out')
+
+
+class TestReduceSumWithNumelOne(OpTest):
```
Comment (wanghaoshuang) [inline python/paddle/fluid/tests/unittests/test_reduce_op.py:263]: 应该再加一个test case:
x.shape=(n, 1)
dim=[1]

File: paddle/fluid/operators/cub_reduce.h (hunk 1 above)
Comment (qingqing01) [reply paddle/fluid/operators/cub_reduce.h:299]: Done. Thanks!

**Review by wanghaoshuang — APPROVED** [review]

**Review by sneaxiy — APPROVED** [review]
Cool

**Pull request merged**